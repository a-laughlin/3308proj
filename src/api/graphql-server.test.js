const { createTestClient } = require('apollo-server-testing');
const { gql } = require('apollo-server-express');
const server = require('./graphql-server');

// we don't need to create a test version of the server or mock outside data
// sources since our implementation is simple
// unless we wanted to test prediction responses without calling the ML piece
// for more info https://www.apollographql.com/docs/apollo-server/features/testing

it('queries nested heartRatePredictions', async () => {
  // create query/mutate fns for the server for the server
  const { query/*, mutate*/ } = createTestClient(server);

  // run query against the server and snapshot the output
  const result = await query({
    query: gql`query ($summary_date: ID!,$steps:Int,$model_id: ID){
      heartRatePredictions (summary_date: $summary_date, steps: $steps, model_id: $model_id){
        summary_date
        bedtime_start
        bedtime_end
        freq
        rates
        prediction_model_id
        history{
          summary_date
          bedtime_start
          bedtime_end
          freq
          rates
          prediction_model_id
        }
      }
    }`,
    variables: { summary_date: "2018-11-05", steps:10 }
  });
  expect(result).toMatchSnapshot();
});
