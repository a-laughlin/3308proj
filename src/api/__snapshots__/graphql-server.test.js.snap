// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queries nested heartRatePredictions 1`] = `
Object {
  "data": Object {
    "heartRatePredictions": Array [
      Object {
        "bedtime_end": "1970-01-01T00:05:30.013Z",
        "bedtime_start": "1970-01-01T00:00:30.013Z",
        "freq": 30000,
        "history": Object {
          "bedtime_end": "13.625",
          "bedtime_start": "81",
          "freq": 30000,
          "prediction_model_id": null,
          "rates": Array [
            80,
            80,
            79,
            78,
            79,
            80,
            80,
            78,
            79,
            79,
          ],
          "summary_date": "2018-11-05",
        },
        "prediction_model_id": "ml_model_foo",
        "rates": Array [
          77,
          77,
          76,
          76,
          76,
          76,
          76,
          76,
          76,
          76,
          76,
        ],
        "summary_date": "2018-11-05_ml_model_foo",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
